gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -520,6 +482,117 @@ private void cleanupStagingData(JobState jobState) {
     }
   }
 
+  /**
+   * Run a given list of {@link WorkUnit}s of a job.
+   *
+   * <p>
+   *   This method calls {@link #runWorkUnits(String, List, TaskStateTracker, TaskExecutor, CountDownLatch)}
+   *   to actually run the {@link Task}s of the {@link WorkUnit}s.
+   * </p>
+   *
+   * @param jobId the job ID
+   * @param workUnits the given list of {@link WorkUnit}s to submit to run
+   * @param taskStateTracker a {@link TaskStateTracker} for task state tracking
+   * @param taskExecutor a {@link TaskExecutor} for task execution
+   * @param taskStateStore a {@link StateStore} for storing {@link TaskState}s
+   * @param logger a {@link Logger} for logging
+   * @throws IOException if there's something wrong with any IO operations
+   * @throws InterruptedException if the task execution gets cancelled
+   */
+  public static void runWorkUnits(String jobId, String containerId, List<WorkUnit> workUnits,
```

Comment:
Actually no. `GobblinHelixTask` and `MRJobLauncher.TaskRunner` use it and both of them do not extend `AbstractJobLauncher`.